### PR TITLE
Append parameters to target URL

### PR DESCRIPTION
**Pull request by jarrettkong**
Implement and test logic to append any parameters to the target URL.

### CODE REVIEW

File: pkg/hook/hook.go [hunk 1]
```diff
@@ -10,9 +11,10 @@ import (
 
 // Hook represents a single hook configuration.
 type Hook struct {
-	Method  string      `yaml:"method"`
-	Headers http.Header `yaml:"headers"`
-	Body    string      `yaml:"body"`
+	Method  string            `yaml:"method"`
+	Headers http.Header       `yaml:"headers"`
+	Body    string            `yaml:"body"`
+	Params  map[string]string `yaml:"params"`
```
Comment (wlynch) [inline pkg/hook/hook.go]: We should use [url.Values](https://godoc.org/net/url#Values) instead. URL params can actually be specified multiple times.

File: pkg/hook/hook.go [hunk 2]
```diff
@@ -55,8 +57,19 @@ func (h *Hook) Dump() ([]byte, error) {
 	return yaml.Marshal(h)
 }
 
+func (h *Hook) joinURLParams() string {
```
Comment (wlynch) [inline pkg/hook/hook.go]: Once you switch to using [url.Values](https://godoc.org/net/url#Values), you can replace this method with `h.Params.Encode()`

File: pkg/hook/hook.go [hunk 3]
```diff
@@ -55,8 +57,19 @@ func (h *Hook) Dump() ([]byte, error) {
...
+	params := make([]string, 0, len(h.Params))
+	for k, v := range h.Params {
+		params = append(params, fmt.Sprintf("%v=%v", k, v))
+	}
+	return "?" + strings.Join(params, "&")
+}
+
 // Fire sends an HTTP request to the given target.
 func (h *Hook) Fire(target string) (*http.Response, error) {
+	if len(h.Params) >= 1 {
```
Comment (wlynch) [inline pkg/hook/hook.go]: Once you switch to using url.Values, you can remove this.

File: pkg/hook/hook.go [hunk 4]
```diff
@@ -55,8 +57,19 @@ func (h *Hook) Dump() ([]byte, error) {
...
+		target += h.joinURLParams()
```
Comment (wlynch) [inline pkg/hook/hook.go]: Set this in `h.toRequest` instead. e.g. something along the lines of https://play.golang.org/p/YgRtsfSVtE5

**Review by wlynch — CHANGES_REQUESTED** [review]

**Comment by eddiezane** [comment]
@jarrettkong Nice job! Would like to see the record logic implemented along with this to close the issue.

**Review by wlynch — CHANGES_REQUESTED** [review]
Looks good! Let's add tests that include parsing both with and without params. In particular, I want to make sure we don't nil panic on the `h.Params` if there are no explicit params defined.

**Comment by jarrettkong** [comment]
@wlynch I was able to test and record the parameters. When they are explicitly defined, the test passes, but unspecified params are failing, even with `reflect.DeepEqual()`. The error is `expected params to be map[] got map[]`, which should pass with `DeepEqual` from my understanding. I also broke the empty params test out into it's own function, as I wasn't sure how to change the params without a new request yet.

**Comment by jarrettkong** [comment]
@eddiezane if I run `hook fire` with the correct args, I get the correct output from record. However, it's automatically changing the `params` in the `test.yml` file from what is defined there to just `{}` and I can't figure out why. After each run I have to go in and undo it manually - very odd.

**Comment by eddiezane** [comment]
@jarrettkong let's rebase with master and take another look

File: cmd/record_test.go [hunk 5]
```diff
@@ -55,48 +55,99 @@ func reqString(t *testing.T, req *http.Request) string {
...
+		}
+
+		srv := httptest.NewServer(r)
+
+		req, err := http.NewRequest(tc.method, srv.URL+tc.query, strings.NewReader(tc.body))
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		for k, v := range tc.headers {
```
Comment (wlynch) [inline cmd/record_test.go]: I think you can replace with with `req.Header = tc.headers`

**Review by wlynch — APPROVED** [review]

File: cmd/record_test.go (hunk 5 above)
Comment (eddiezane) [reply cmd/record_test.go]: Good catch. I was worried that would blow away the default headers but I it makes sense that they are populated last.

**Pull request merged**